docs/app/Component Guidelines.md

```diff
@@ -1,12 +1,21 @@
 # Stardust Component Guidelines
```

Comment:
The updated document best reviewed in rendered form: https://github.com/TechnologyAdvice/stardust/blob/feature/message-component/docs/app/Component%20Guidelines.md